src/app/zap-templates/zcl/data-model/chip/service-area-cluster.xml

```diff
@@ -0,0 +1,131 @@
+    <item name="PositionTag"       type="PositionTag"           optional="false" isNullable="true"/>
+    <item name="SurfaceTag"        type="FloorSurfaceTag"       optional="false" isNullable="true"/>
+  </struct>
+
+  <struct name="MapStruct">
+    <cluster code="0x0150"/>
+    <item name="MapId"   type="int8u"                   optional="false"/>
+    <item name="Name"    type="char_string" length="64" optional="false"/>
+  </struct>
+
+  <struct name="LocationStruct">
+    <cluster code="0x0150"/>
+    <item name="LocationId"   type="int32u"                  optional="false" isNullable="false"/>
+    <item name="MapId"        type="int8u"                   optional="false" isNullable="true"/>
+    <item name="LocationInfo" type="LocationInfoStruct"      optional="false" isNullable="false"/>
+  </struct>
+
+  <struct name="ProgressStruct">
+    <cluster code="0x0150"/>
+    <item name="LocationId"             type="int32u"                 optional="false"/>
+    <item name="Status"                 type="OperationalStatusEnum"  optional="false"/>
+    <item name="TotalOperationalTime"   type="elapsed_s"              optional="true" isNullable="true"/>
+    <item name="EstimatedTime"          type="elapsed_s"              optional="true" isNullable="true"/>
+  </struct>
+
+  <!-- Enum definitions for select locations command response-->
+  <enum name="SelectLocationsStatus" type="enum8">
+    <cluster code="0x0150"/>
+    <item value="0x00" name="Success"/>
+    <item value="0x01" name="UnsupportedLocation"/>
+    <item value="0x02" name="DuplicatedLocations"/>
+    <item value="0x03" name="InvalidInMode"/>
+    <item value="0x04" name="InvalidSet"/>
+  </enum>
+
+  <!-- Enum definitions for skip location command response-->
+  <enum name="SkipCurrentStatus" type="enum8">
+    <cluster code="0x0150"/>
+    <item value="0x00" name="Success"/>
+    <item value="0x01" name="InvalidLocationList"/>
+    <item value="0x02" name="InvalidInMode"/>
+  </enum>
+
+
+  <cluster>
+    <domain>General</domain>
+    <name>Service Area</name>
+      <code>0x0150</code>
+    <define>SERVICE_AREA_CLUSTER</define>
+    <description>The Service Area cluster provides an interface for controlling the locations where a device should operate, and for querying the current location.</description>
+
+    <!-- Attributes -->
+    <attribute side="server" code="0x0000" define="SupportedLocations"  type="array"     entryType="LocationStruct"  writable="false" isNullable="false" optional="false">SupportedLocations</attribute>
+    <attribute side="server" code="0x0001" define="SupportedMaps"       type="array"     entryType="MapStruct"       writable="false" isNullable="true"  optional="false">SupportedMaps</attribute>
+    <attribute side="server" code="0x0002" define="SelectedLocations"   type="array"     entryType="int32u"          writable="false" isNullable="true"  optional="false" scene="true">SelectedLocations</attribute>
```

Comment:
Why does this have scene="true"?

Reply:
@johnfierke ?

Reply:
From the spec:
=== Scene Table Extensions

If the Scenes Management server cluster is implemented on the same endpoint, the following
attribute SHALL be part of the ExtensionFieldSetStruct of the Scene Table.

* SelectedLocations

Reply:
@johnfierke I don't see that text in the spec as of rev b3652909bdf595c7634cea1da0f1e58f837ea35f.  It looks like it was there in some in-progress versions of https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8937, but was removed in https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8937/commits/eaabc9964052dbf78f977c266aa859b19dabef3e (and the Scenable quality was removed in https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8937/commits/f2e95c9061cc21ac55f5ed01ba1e7bf96cfd7bc9).

It would be really good if when we say "from the spec" we were all looking at the same spec.

Reply:
The spec has been changing daily. The xml defintions have fallen behind. Not sure what we can do except try to update ASAP and hope the spec stabilizes soon.

Most if the changes are names and editorial, so they don't have huge impact on implementation, but it's still annoying when they aren't in sync.

Reply:
This has now been removed.